ironfish/src/chainProcessor.ts

```diff
@@ -96,20 +96,40 @@ export class ChainProcessor {
       }
     }
 
-    const iter = this.chain.iterateTo(fork, chainHead, undefined, false)
+    if (chainHead.sequence < head.sequence) {
```

Comment:
Why cant you just change the `isLinear` above to this condition?

Reply:
`chainHead` could still be ahead of `head` and you'd need to do the block above. Suppose `chainHead` is chain A and `head` is chain B:
```
A1 -> A2 -> A3 -> A4
               -> B3
```
`head` needs to go backwards from B3 -> A2, and then forwards from A2 -> A3 -> A4

Reply:
I think we might be able to change it to `if (fork.sequence < head.sequence)`. The block below seems to assume `fork === head`

Reply:
Sorry, yes. That's what I mean. fork.sequence < head.sequence

Reply:
Hugh was saying we can always just rewind to the fork then fast-forward to the destination and that should handle all cases. Seems sound. Added a comment here for explanation